packages/slate-react/src/utils/clone-fragment.js

```diff
@@ -107,31 +110,30 @@ function cloneFragment(event, value, fragment = value.fragment) {
 
   attach.setAttribute('data-slate-fragment', encoded)
 
-  // Add the phony content to the DOM, and select it, so it will be copied.
-  const editor = event.target.closest('[data-slate-editor]')
+  // Add the phony content to a div element. This is needed to copy the
+  // contents into the html clipboard register.
   const div = window.document.createElement('div')
+  div.appendChild(contents)
+
+  // For browsers supporting it, we set the clipboard registers manually,
+  // since the result is more predictable.
+  if (event.clipboardData && event.clipboardData.setData) {
+    event.preventDefault()
+    event.clipboardData.setData(TEXT, native.toString())
+    event.clipboardData.setData(FRAGMENT, encoded)
+    event.clipboardData.setData(HTML, div.innerHTML)
+    return
+  }
+
+  // COMPAT: For browser that don't support the Clipboard API's setData method,
+  // we must rely on the browser to natively copy what's selected.
+  // So we add the div (containing our content) to the DOM, and select it.
+  const editor = event.target.closest('[data-slate-editor]')
   div.setAttribute('contenteditable', true)
   div.style.position = 'absolute'
   div.style.left = '-9999px'
-
-  // COMPAT: In Firefox, the viewport jumps to find the phony div, so it
```

Comment:
Removed this since Firefox support `clipboardData.setData()`.

Reply:
@tobiasandersen I noticed that empty blocks now contains a "zero-width space" character when copied and pasted in a notepad.

Replacing the highlighted line with the following effectively remove the zero-width spaces - but there's probably a better fix, I'm not familiar with this problem.
`event.clipboardData.setData(TEXT, native.toString().replace(/\u200B/g, ''))`